shell/platform/windows/window_win32_unittests.cc

```diff
@@ -231,5 +231,29 @@ TEST(MockWin32Window, KeyDownWithCtrlToggled) {
   SetKeyboardState(keyboard_state);
 }
 
+TEST(MockWin32Window, WindowOnMinimized) {
+  MockWin32Window window;
+  window.InjectWindowMessage(WM_SIZE, SIZE_RESTORED, MAKEWPARAM(1, 1));
+
+  EXPECT_CALL(window, OnMinimized()).Times(1);
+  window.InjectWindowMessage(WM_SIZE, SIZE_MINIMIZED, MAKEWPARAM(0, 0));
+
+  // notify only when window size is different.
+  EXPECT_CALL(window, OnMinimized()).Times(0);
+  window.InjectWindowMessage(WM_SIZE, SIZE_MINIMIZED, MAKEWPARAM(0, 0));
+}
+
+TEST(MockWin32Window, WindowOnRestoredFromMinimized) {
+  MockWin32Window window;
+  window.InjectWindowMessage(WM_SIZE, SIZE_MINIMIZED, MAKEWPARAM(0, 0));
+
+  EXPECT_CALL(window, OnRestoredFromMinimized()).Times(1);
+  window.InjectWindowMessage(WM_SIZE, SIZE_RESTORED, MAKEWPARAM(1, 1));
+
+  // notify only when window size is different.
+  EXPECT_CALL(window, OnRestoredFromMinimized()).Times(0);
+  window.InjectWindowMessage(WM_SIZE, SIZE_RESTORED, MAKEWPARAM(1, 1));
```

Comment:
Same comment as before: this should only be notified when the resizing type changes from `SIZE_MINIMIZED` to `SIZE_RESTORED` or `SIZE_MAXIMIZED` right?

If so, consider:

```suggestion
  EXPECT_CALL(window, OnRestoredFromMinimized()).Times(1);
  window.InjectWindowMessage(WM_SIZE, SIZE_RESTORED, MAKEWPARAM(1, 1));

  // Notify only when resizing type changes from minimized.
  EXPECT_CALL(window, OnRestoredFromMinimized()).Times(0);
  window.InjectWindowMessage(WM_SIZE, SIZE_RESTORED, MAKEWPARAM(2, 2));
    window.InjectWindowMessage(WM_SIZE, SIZE_MAXIMIZED, MAKEWPARAM(3, 3));
```

Reply:
It looks nice! I applied this change. I'll mark this as resolved.